fhir-config/src/main/java/com/ibm/fhir/config/ResourcesConfigAdapter.java

```diff
@@ -0,0 +1,112 @@
+
+    public ResourcesConfigAdapter(PropertyGroup resourcesConfig, FHIRVersionParam fhirVersion) throws Exception {
+        supportedTypes = computeSupportedResourceTypes(resourcesConfig, fhirVersion);
+
+        if (resourcesConfig == null) {
+            for (Interaction interaction : Interaction.values()) {
+                typesByInteraction.put(interaction, supportedTypes);
+            }
+            return;
+        }
+
+        for (String resourceType : supportedTypes) {
+            List<String> interactions = resourcesConfig.getStringListProperty(resourceType + "/" + FHIRConfiguration.PROPERTY_FIELD_RESOURCES_INTERACTIONS);
+            if (interactions == null) {
+                interactions = resourcesConfig.getStringListProperty("Resource/" + FHIRConfiguration.PROPERTY_FIELD_RESOURCES_INTERACTIONS);
+            }
+
+            if (interactions == null) {
+                for (Interaction interaction : Interaction.values()) {
+                    typesByInteraction.computeIfAbsent(interaction, k -> new LinkedHashSet<>()).add(resourceType);
+                }
+                continue;
+            }
+
+            for (String interactionString : interactions) {
+                Interaction interaction = Interaction.from(interactionString);
+                typesByInteraction.computeIfAbsent(interaction, k -> new LinkedHashSet<>()).add(resourceType);
+            }
+        }
+    }
+
+    /**
+     * @return an immutable, non-null set of supported resource types for the given fhirVersion
+     * @throws Exception
+     */
+    public Set<String> getSupportedResourceTypes() {
+        return supportedTypes;
+    }
+
+    /**
+     * @return an immutable, non-null set of resource types that are configured for the given interaction and fhirVersion
+     */
+    public Set<String> getSupportedResourceTypes(Interaction interaction) {
+        return typesByInteraction.get(interaction);
+    }
+
+    /**
+     * Construct the list of supported resource types from the passed configuration and fhirVersion
+     *
+     * @param resourcesConfig
+     * @param fhirVersion
+     * @return
+     * @throws Exception
+     */
+    private Set<String> computeSupportedResourceTypes(PropertyGroup resourcesConfig, FHIRVersionParam fhirVersion) throws Exception {
+        Set<String> applicableTypes = ResourceTypeHelper.getResourceTypesFor(fhirVersion);
+
+        if (resourcesConfig == null || resourcesConfig.getBooleanProperty("open", true)) {
+            return applicableTypes;
+        }
+
+        Set<String> result = new LinkedHashSet<String>();
+        for (PropertyEntry rsrcsEntry : resourcesConfig.getProperties()) {
+            String name = rsrcsEntry.getName();
+
+            // Ensure we skip over the special property "open"
+            // and skip the abstract types Resource and DomainResource
+            if (FHIRConfiguration.PROPERTY_FIELD_RESOURCES_OPEN.equals(name) ||
+                    "Resource".equals(name) ||
```

Comment:
Do we have these as constants anywhere? We use them in a few places now.

Reply:
thinking of changing it to `ResourceTypeHelper.getAbstractResourceTypeNames().contains(name)`

Reply:
done